crypto/signature.go

```diff
@@ -81,20 +89,19 @@ func processSignatureExpiration(md *openpgp.MessageDetails, verifyTime int64) {
 
 // verifyDetailsSignature verifies signature from message details.
 func verifyDetailsSignature(md *openpgp.MessageDetails, verifierKey *KeyRing) error {
-	if md.IsSigned {
-		if md.SignedBy == nil || len(verifierKey.entities) == 0 {
-			return newSignatureNoVerifier()
-		}
-		matches := verifierKey.entities.KeysById(md.SignedByKeyId)
-		if len(matches) > 0 {
-			if md.SignatureError == nil {
-				return nil
-			}
-			return newSignatureFailed()
-		}
+	if !md.IsSigned ||
+		md.SignedBy == nil ||
+		len(verifierKey.entities) == 0 ||
+		len(verifierKey.entities.KeysById(md.SignedByKeyId)) == 0 {
+		return newSignatureNoVerifier()
 	}
-
-	return newSignatureNoVerifier()
+	if md.SignatureError != nil ||
+		md.Signature == nil ||
+		md.Signature.Hash < allowedHashes[0] ||
+		md.Signature.Hash > allowedHashes[len(allowedHashes)-1] {
```

Comment:
This depends on the order of the hash listing on `x/crypto`. Not that it's going to change anytime soon but I would simply check against the `allowedHashes` global variable introduced in this commit, so that adding/removing hashes will be as simple as modifying it.

Reply:
Yeah. I kind of agree with you but this would introduce an extra loop (as there is no built-in function to check for inclusion in an array) and make this code roughly three times bigger. I think we can do that if we ever want to change `allowedHashes` to be a non-continuous range (which I find somewhat unlikely as usually the disallowed ones will be the weakest).